internal/controllers/core/clustercatalog_controller.go

```diff
@@ -180,73 +176,60 @@ func (r *ClusterCatalogReconciler) reconcile(ctx context.Context, catalog *v1alp
 
 		return ctrl.Result{RequeueAfter: requeueAfter}, nil
 	default:
-		return ctrl.Result{}, updateStatusUnpackFailing(&catalog.Status, fmt.Errorf("unknown unpack state %q: %v", unpackResult.State, err))
+		panic(fmt.Sprintf("unknown unpack state %q", unpackResult.State))
 	}
 }
 
-func updateStatusUnpackPending(status *v1alpha1.ClusterCatalogStatus, result *source.Result) {
-	status.ResolvedSource = nil
-	meta.SetStatusCondition(&status.Conditions, metav1.Condition{
-		Type:    v1alpha1.TypeUnpacked,
-		Status:  metav1.ConditionFalse,
-		Reason:  v1alpha1.ReasonUnpackPending,
-		Message: result.Message,
-	})
-}
-
-func updateStatusUnpacking(status *v1alpha1.ClusterCatalogStatus, result *source.Result) {
-	status.ResolvedSource = nil
-	meta.SetStatusCondition(&status.Conditions, metav1.Condition{
-		Type:    v1alpha1.TypeUnpacked,
-		Status:  metav1.ConditionFalse,
-		Reason:  v1alpha1.ReasonUnpacking,
-		Message: result.Message,
-	})
+func updateStatusProgressing(catalog *v1alpha1.ClusterCatalog, err error) error {
+	var unrecov *catalogderrors.Unrecoverable
+	if err == nil {
+		catalog.Status.ResolvedSource = nil
+		meta.SetStatusCondition(&catalog.Status.Conditions, metav1.Condition{
+			Type:    v1alpha1.TypeProgressing,
+			Status:  metav1.ConditionFalse,
+			Reason:  v1alpha1.ReasonSucceeded,
+			Message: fmt.Sprintf("Successfully unpacked and stored content from %s", catalog.Spec.Source.Image.Ref),
+		})
+	} else if errors.As(err, &unrecov) {
+		meta.SetStatusCondition(&catalog.Status.Conditions, metav1.Condition{
+			Type:    v1alpha1.TypeProgressing,
+			Status:  metav1.ConditionFalse,
+			Reason:  v1alpha1.ReasonUnrecoverable,
+			Message: err.Error(),
+		})
+		return err
+	} else {
+		meta.SetStatusCondition(&catalog.Status.Conditions, metav1.Condition{
+			Type:    v1alpha1.TypeProgressing,
+			Status:  metav1.ConditionTrue,
+			Reason:  v1alpha1.ReasonRetrying,
+			Message: err.Error(),
+		})
+		return err
+	}
+	return nil
```

Comment:
Missed this in my last review round, but would returning early make sense here instead of this longer if-elseif-else chain?
```suggestion
	var unrecov *catalogderrors.Unrecoverable
        if errors.As(err, &unrecov) {
		meta.SetStatusCondition(&catalog.Status.Conditions, metav1.Condition{
			Type:    v1alpha1.TypeProgressing,
			Status:  metav1.ConditionFalse,
			Reason:  v1alpha1.ReasonUnrecoverable,
			Message: err.Error(),
		})
		return err
	}
	
	if err != nil {
		meta.SetStatusCondition(&catalog.Status.Conditions, metav1.Condition{
			Type:    v1alpha1.TypeProgressing,
			Status:  metav1.ConditionTrue,
			Reason:  v1alpha1.ReasonRetrying,
			Message: err.Error(),
		})
		return err
	}
	
	catalog.Status.ResolvedSource = nil
	meta.SetStatusCondition(&catalog.Status.Conditions, metav1.Condition{
		Type:    v1alpha1.TypeProgressing,
		Status:  metav1.ConditionFalse,
		Reason:  v1alpha1.ReasonSucceeded,
		Message: fmt.Sprintf("Successfully unpacked and stored content from %s", catalog.Spec.Source.Image.Ref),
	})
	return nil
```